src/main/java/wooteco/subway/service/MemberService.java

```diff
@@ -0,0 +1,48 @@
+package wooteco.subway.service;
+
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import wooteco.subway.dao.MemberDao;
+import wooteco.subway.domain.Member;
+import wooteco.subway.dto.MemberRequest;
+import wooteco.subway.dto.MemberResponse;
+
+@Service
+public class MemberService {
+    private MemberDao memberDao;
+    private PasswordEncoder passwordEncoder;
+
+    public MemberService(MemberDao memberDao, PasswordEncoder passwordEncoder) {
+        this.memberDao = memberDao;
+        this.passwordEncoder = passwordEncoder;
+    }
+
+    public MemberResponse createMember(MemberRequest request) {
+        String email = request.getEmail();
+        String password = passwordEncoder.encode(request.getPassword());
+        Integer age = request.getAge();
+        Member member = memberDao.insert(new Member(email, password, age));
+        return MemberResponse.of(member);
+    }
+
+    public MemberResponse findMember(Long id) {
+        Member member = memberDao.findById(id);
+        return MemberResponse.of(member);
+    }
+
+    public void updateMember(Long id, MemberRequest memberRequest) {
+        memberDao.update(new Member(id, memberRequest.getEmail(), memberRequest.getPassword(), memberRequest.getAge()));
+    }
+
+    public void deleteMember(Long id) {
+        memberDao.deleteById(id);
+    }
+
+    public MemberResponse logIn(String email, String password) {
+        Member member = memberDao.findByEmail(email).orElseThrow(() -> new IllegalArgumentException("이메일을 잘못 입력하셨습니다."));
+        if (passwordEncoder.matches(password, member.getPassword())) {
+            return MemberResponse.of(member);
+        }
+        throw new IllegalArgumentException("비밀번호가 맞지 않습니다.");
+    }
```

Comment:
```suggestion
    public MemberResponse logIn(String email, String password) {
        Member member = memberDao.findByEmail(email)
            .orElseThrow(() -> new IllegalArgumentException("이메일을 잘못 입력하셨습니다."));
        
        if (passwordEncoder.matches(password, member.getPassword())) {
            return MemberResponse.of(member);
        }
        throw new IllegalArgumentException("비밀번호가 맞지 않습니다.");
    }
```


~~member 객체에게 패스워드가 맞는지 물어보는 방법은 없을까요?~~

Reply:
찾아보니 BCryptPasswordEncoder 구현체가 매번 다른 값을 리턴하군요! 👍

Reply:
로그인에 대한 예외메세지도 고민해보면 좋을 것 같아요 

이메일이 입력이 잘못되었다는 메세지는 해당 이메일에 대한 정보는 없다는 뜻이고 
패스워드가 틀렸다는 것은 이메일은 있지만 패스워드는 틀렸다는 것을 의미합니다! 
그렇다면 공격의 범위가 축소된 것 아닐까요?

Reply:
아하!! 그래서 요즘 웹서비스가 비밀번호든 이메일이든 무엇이 틀려도 같은 메세지를 출력해주는 추세였던 거군요..! 좋은 피드백 감사합니다. [반영](https://github.com/tributetothemoon/atdd-subway-path/commit/808ae180e1f5d896e871f77a32da4e6b0385da92)해보았습니다!